docs/dev/design/redirects.rst

```diff
@@ -0,0 +1,338 @@
+   is created, it will redirect all paths that start with ``/en/latest/dir/``,
+   the rest of the path will be added to the new URL automatically.
+
+   - Cross domain redirects are allowed in exact redirects.
+   - They apply to all versions.
+   - A wildcard is allowed at the end of the URL.
+   - If a wildcard is used, the rest of the path will be added to the new URL automatically.
+
+Sphinx HTMLDir to HTML:
+   Allows to redirect clean-URLs to HTML URLs.
+   Useful in case a project changed the style of their URLs.
+
+   They apply to all projects, not just Sphinx projects.
+
+Sphinx HTML to HTMLDir:
+   Allows to redirect HTML URLs to clean-URLs.
+   Useful in case a project changed the style of their URLs.
+
+   They apply to all projects, not just Sphinx projects.
+
+How other services implement redirects
+--------------------------------------
+
+- Gitbook implementation is very basic,
+  they only allow page redirects.
+
+  https://docs.gitbook.com/integrations/git-sync/content-configuration#redirects
+
+- Cloudflare pages allow to capture placeholders and one wildcard (in any part of the URL).
+  They also allow you to set the status code of the redirect,
+  and redirects can be specific in a ``_redirects`` file.
+
+  https://developers.cloudflare.com/pages/platform/redirects/
+
+  They have a limit of 2100 redirects.
+  In case of multiple matches, the topmost redirect will be used.
+
+- Netlify allows to capture placeholders and a wildcard (only allowed at the end).
+  They also allow you to set the status code of the redirect,
+  and redirects can be specific in a ``_redirects`` file.
+
+  - Forced redirects
+  - Match query arguments
+  - Match by country/language and cookies
+  - Per-domain and protocol redirects
+  - In case of multiple matches, the topmost redirect will be used.
+
+  https://docs.netlify.com/routing/redirects/
+
+Improvements
+------------
+
+General improvements
+~~~~~~~~~~~~~~~~~~~~
+
+The following improvements will be applied to all types of redirects.
+
+- Allow choosing the status code of the redirect.
+  We already have a field for this, but it's not exposed to users.
+- Allow to explicitly define the order of redirects.
+  This will be similar to the automation rules feature,
+  where users can reorder the rules so the most specific ones are first.
+  We currently rely on the implicit order of the redirects (updated_at).
+- Allow to disable redirects.
+  It's useful when testing redirects, or when debugging a problem.
+  Instead of having to re-create the redirect,
+  we can just disable it and re-enable it later.
+- Allow to add a short description.
+  It's useful to document why the redirect was created.
+
+Don't run redirects on domains from pull request previews
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We currently run redirects on domains from pull request previews,
+this is a problem when moving a whole project to a new domain.
```

Comment:
I'm not sure to understand what it the problem here. Could you expand a little more on this?

Reply:
If you want to redirect your old domains to a new domain, you can use the trick from

https://docs.readthedocs.io/en/stable/user-defined-redirects.html#migrating-your-documentation-to-another-domain

There is no problems when the domain you're moving away to isn't hosted on RTD, but if it's hosted on RTD, you won't be able to use PR previews, since that domain will also redirect to the other one.

See https://github.com/readthedocs/readthedocs.org/pull/10825#discussion_r1367140568 and https://github.com/readthedocs/readthedocs.org/issues/9614